source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_resource_detector.cc

```diff
@@ -0,0 +1,57 @@
+#include "source/extensions/tracers/opentelemetry/resource_detectors/dynatrace/dynatrace_resource_detector.h"
+
+#include <fstream>
+#include <iostream>
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+namespace {
+
+void addAttributes(const std::string& content, Resource& resource) {
+  for (const auto& line : StringUtil::splitToken(content, "\n", false, true)) {
+    const auto keyValue = StringUtil::splitToken(line, "=");
+    if (keyValue.size() != 2) {
+      continue;
+    }
+
+    const std::string key = std::string(keyValue[0]);
+    const std::string value = std::string(keyValue[1]);
+    resource.attributes_[key] = value;
+  }
+}
+
+} // namespace
+
+Resource DynatraceResourceDetector::detect() {
+  Resource resource;
+  resource.schemaUrl_ = "";
+  int failureCount = 0;
+
+  for (const auto& file_name : DT_METADATA_FILES) {
+    TRY_NEEDS_AUDIT {
+      std::string content = dynatrace_file_reader_->readEnrichmentFile(std::string(file_name));
```

Comment:
use CONSTRUCT_ON_FIRST_USE for `DT_METADATA_FILES` then you don't need to stringify file_name everytime.

Reply:
I changed it, let me know if that's what you meant. [446a31d](https://github.com/envoyproxy/envoy/pull/30824/commits/446a31de3e2a01fd1ff64d74866b39ddef0bbd6b)

Reply:
Yes that lgtm